charts/fluent-bit/templates/daemonset.yaml

```diff
@@ -0,0 +1,92 @@
+---
+{{- $scope := . }}
+{{- $values := .Values }}
+{{- $release := .Release }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ include "chart.name" $scope }}
+  namespace: {{ $release.Namespace }}
+  labels:
+{{ include "chart.labels" $scope | indent 4 }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ include "chart.name" $scope }}
+      release: {{ $release.Name }}
+      namespace: {{ $release.Namespace }}
+      version: {{ include "chart.appVersion" $scope }}
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 1
+  template:
+    metadata:
+      annotations:
+        # In sprig `false` as a boolean is considered empty so the value of the following tag will be always true
+        prometheus.io/scrape: {{ printf "'%s'" (default "true" $values.prometheus.enabled | toString) }}
+        prometheus.io/path: {{ default "/metrics" $values.prometheus.path }}
+        prometheus.io/port: {{ printf "'%s'" (default "8082" $values.prometheus.port | toString ) }}
+        prometheus.io/scheme: {{ default "http" $values.prometheus.scheme }}
+      labels:
+        app: {{ include "chart.name" $scope }}
+        release: {{ $release.Name }}
+        namespace: {{ $release.Namespace }}
+        version: {{ include "chart.appVersion" $scope }}
+    spec:
+      serviceAccountName: {{ include "chart.serviceAccountName" $scope }}
+      securityContext:
+{{ toYaml $values.podSecurityContext | indent 8 }}
+      containers:
+        - name: service
+          securityContext:
+{{ toYaml $values.securityContext | indent 12 }}
+          env:
+          image: {{ printf "%s/%s:%s" $values.image.registry $values.image.repository $values.image.tag }}
+          imagePullPolicy: {{ default $values.image.pullPolicy "IfNotPresent" }}
+          ports:
+  {{- if and $values.prometheus.enabled }}
+            - name: metrics
```

Comment:
If the purpose of exposing this container port is only for metric collection , we don't need to expose it.